pyvista/utilities/fileio.py

```diff
@@ -210,39 +225,23 @@ def read(filename, attrs=None, file_format=None):
     filename = os.path.abspath(os.path.expanduser(str(filename)))
     if not os.path.isfile(filename):
         raise FileNotFoundError(f'File ({filename}) not found')
-    ext = get_ext(filename)
+    if force_ext:
+        ext = str(force_ext).lower()
+    else:
+        ext = get_ext(filename)
 
     # Read file using meshio.read if file_format is present
     if file_format:
         return read_meshio(filename, file_format)
 
     # From the extension, decide which reader to use
     if attrs is not None:
-        reader = get_reader(filename)
+        reader = get_reader(filename, force_ext=ext)
         return standard_reader_routine(reader, filename, attrs=attrs)
-    elif ext in '.vti': # ImageData
-        return pyvista.UniformGrid(filename)
-    elif ext in '.vtr': # RectilinearGrid
-        return pyvista.RectilinearGrid(filename)
-    elif ext in '.vtu': # UnstructuredGrid
-        return pyvista.UnstructuredGrid(filename)
-    elif ext in ['.ply', '.obj', '.stl']: # PolyData
-        return pyvista.PolyData(filename)
-    elif ext in '.vts': # StructuredGrid
-        return pyvista.StructuredGrid(filename)
-    elif ext in ['.vtm', '.vtmb', '.case']:
-        return pyvista.MultiBlock(filename)
-    elif ext in ['.e', '.exo']:
-        return read_exodus(filename)
-    elif ext in ['.vtk']:
-        # Attempt to use the legacy reader...
-        return read_legacy(filename)
```

Comment:
`read_legacy` seems to be no longer reachable?

Reply:
Woops, added these back with tests.